examples/benchmarks/contextvalues_test.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2022 The logr Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package benchmarks
+
+import (
+	"context"
+	"testing"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/logr/funcr"
+)
+
+const iterationsPerOp = 100
+
+// 1% of the Info calls are invoked, all of those call the LogSink.
+func BenchmarkNewContext1Percent(b *testing.B) {
+	ctx := logr.NewContext(context.Background(), discardWithV(b, 1))
+	defer func() {
+		expected := int64(iterationsPerOp) / 100 * int64(b.N)
+		if calls != expected {
+			b.Errorf("expected %d calls to Info, got %d", expected, calls)
+		}
+		calls = 0
```

Comment:
why not make calls local and pass it by pointer into `discardWithV` ?

Reply:
Why indeed? I simplified this by moving all of the common code into a `setup` function.